app/controllers/api/v2/fact_values_controller.rb

```diff
@@ -18,6 +18,10 @@ def index
         host_id = params[:host_id]
         @fact_values = FactValue.build_facts_hash(values, host_id)
       end
+
+      def setup_search_options
+        params[:search] = " host = " + params[:host_id] if params[:host_id]
```

Comment:
this should be added to the search param, not override it. also keep in mind the param might not be there.